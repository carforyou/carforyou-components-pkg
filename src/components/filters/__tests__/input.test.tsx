import React from "react"
import { fireEvent, render } from "@testing-library/react"

import InputFilter from "../input"

describe("<InputFilter>", () => {
  describe("renders correctly", () => {
    it("with no values", () => {
      const { container } = render(
        <InputFilter
          mode="numeric"
          name="test"
          placeholder="test"
          apply={jest.fn()}
        />
      )

      expect(container).toMatchSnapshot()
    })

    it("with autofocus after focusing", () => {
      const { container, getByPlaceholderText } = render(
        <InputFilter
          mode="numeric"
          name="test"
          placeholder="test"
          apply={jest.fn()}
        />
      )

      fireEvent.focus(getByPlaceholderText("test"))

      expect(container).toMatchSnapshot()
    })
  })
})
