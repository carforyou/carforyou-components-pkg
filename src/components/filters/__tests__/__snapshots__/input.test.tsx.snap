// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputFilter> renders correctly with autofocus after focusing 1`] = `
<div>
  <div
    class="w-12/12 focus-within:text-teal"
  >
    <div
      class="relative pb-px"
    >
      <div
        class="validation-inputContainer mb-20"
      >
        <div
          class="relative flex w-12/12 clearButton-container"
        >
          <input
            autofocus=""
            class="w-12/12 input_withClearButton"
            data-testid="test"
            data-valid="true"
            id="test"
            inputmode="numeric"
            name="test"
            placeholder="test"
            type="number"
            value=""
          />
        </div>
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`<InputFilter> renders correctly with no values 1`] = `
<div>
  <div
    class="w-12/12 focus-within:text-teal"
  >
    <div
      class="relative pb-px"
    >
      <div
        class="validation-inputContainer mb-20"
      >
        <div
          class="relative flex w-12/12 clearButton-container"
        >
          <input
            class="w-12/12 input_withClearButton"
            data-testid="test"
            data-valid="true"
            id="test"
            inputmode="numeric"
            name="test"
            placeholder="test"
            type="number"
            value=""
          />
        </div>
      </div>
      <div />
    </div>
  </div>
</div>
`;
